### PR TITLE
media-sound/whipper: bump version to 0.9.0

### PR DESCRIPTION
**Pull request by CaseOf**
This new version of whipper now support Python 3.

### CODE REVIEW

File: media-sound/whipper/whipper-0.9.0.ebuild [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+inherit distutils-r1
+
+DESCRIPTION="A Python CD-DA ripper preferring accuracy over speed (forked from morituri)"
+HOMEPAGE="https://github.com/whipper-team/whipper"
+SRC_URI="https://github.com/whipper-team/${PN}/archive/v${PV}.tar.gz -> ${P}.tar.gz"
+
+LICENSE="GPL-3+"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="test"
```
Comment (a17r) [inline media-sound/whipper/whipper-0.9.0.ebuild]: missing `RESTRICT+=" !test? ( test )"` then.

File: media-sound/whipper/whipper-0.9.0.ebuild [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+DEPEND="
+	dev-python/setuptools[${PYTHON_USEDEP}]
+	media-libs/libsndfile
```
[the remark this answers is not in the document]
Comment (CaseOf) [reply media-sound/whipper/whipper-0.9.0.ebuild]: without libsndfile, build fails.

File: media-sound/whipper/whipper-0.9.0.ebuild [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+	media-libs/mutagen[${PYTHON_USEDEP}]
+	media-sound/sox[flac]
+"
+
+PATCHES=(
+	"${FILESDIR}/${PN}-0.7.0-cdparanoia-name-fix.patch"
+	"${FILESDIR}/${PN}-0.7.0-src-Makefile-respect-CFLAGS.patch"
+)
+
+src_prepare() {
```
Comment (a17r) [inline media-sound/whipper/whipper-0.9.0.ebuild]: last time I was working on a python ebuild I was told to make this `python_prepare_all`

File: media-sound/whipper/whipper-0.9.0.ebuild [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+	# accurip test totally depends on network access
+	rm "${PN}"/test/test_common_accurip.py || die
+
+	distutils-r1_src_prepare
```
Comment (a17r) [inline media-sound/whipper/whipper-0.9.0.ebuild]: and consequently, this `distutils-r1_python_prepare_all`

File: media-sound/whipper/whipper-0.9.0.ebuild (hunk 2 above)
[the remark this answers is not in the document]
Comment (a17r) [reply media-sound/whipper/whipper-0.9.0.ebuild]: if yes, put it in `COMMON_DEPEND` instead and inherit this in both `DEPEND` and `RDEPEND` instead.

File: media-sound/whipper/whipper-0.9.0.ebuild [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+DEPEND="
+	dev-python/setuptools[${PYTHON_USEDEP}]
+	media-libs/libsndfile
+	test? ( dev-python/twisted[${PYTHON_USEDEP}] )
+"
+RDEPEND="
+	app-cdr/cdrdao
+	>=dev-libs/libcdio-paranoia-0.94_p2
+	>=dev-python/pycdio-2.1.0[${PYTHON_USEDEP}]
+	dev-python/pygobject:3=[${PYTHON_USEDEP}]
```
Comment (a17r) [inline media-sound/whipper/whipper-0.9.0.ebuild]: that slot operator looks really weird. there is no subslot on this package so just drop it

File: media-sound/whipper/whipper-0.9.0.ebuild [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+
+BDEPEND="test? ( dev-python/twisted[${PYTHON_USEDEP}] )"
+COMMON_DEPEND="media-libs/libsndfile"
+DEPEND="${COMMON_DEPEND}"
+RDEPEND="
+	${COMMON_DEPEND}
+	app-cdr/cdrdao
+	>=dev-libs/libcdio-paranoia-0.94_p2
+	>=dev-python/pycdio-2.1.0[${PYTHON_USEDEP}]
+	dev-python/pygobject:=[${PYTHON_USEDEP}]
```
Comment (a17r) [inline media-sound/whipper/whipper-0.9.0.ebuild]: I was asking to drop the slot op, not the slot dep ;)

File: media-sound/whipper/whipper-0.9.0.ebuild [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+
+LICENSE="GPL-3+"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="test"
+RESTRICT+=" !test? ( test )"
+
+BDEPEND="test? ( dev-python/twisted[${PYTHON_USEDEP}] )"
+COMMON_DEPEND="media-libs/libsndfile"
+DEPEND="${COMMON_DEPEND}"
```
Comment (a17r) [inline media-sound/whipper/whipper-0.9.0.ebuild]: so now that `DEPEND` is empty except for libsndfile, you can drop the `COMMON_DEPEND` and have `RDEPEND` inherit `DEPEND` instead. ;)

**Comment by a17r** [comment]
Sorry for accidental push, reverted.

`whipper-0.7.0-src-Makefile-respect-CFLAGS.patch` failed to apply, did you test this?

**Pull request closed without merging**